core/src/test/scala/caliban/RenderingSpec.scala

```diff
@@ -0,0 +1,62 @@
+package caliban
+
+import zio.test._
+import Assertion._
+import TestUtils._
+import caliban.GraphQL._
+
+object RenderingSpec
+    extends DefaultRunnableSpec(
+      suite("rendering")(
+        test("it should render directives") {
+          assert(
+            graphQL(resolver).render.trim,
+            equalTo("""
+                      |union Role = Captain | Engineer | Mechanic | Pilot
```

Comment:
To have a stable order, we could sort the types by kind, then by name.

Reply:
Thats a good idea I'll give that a shot.